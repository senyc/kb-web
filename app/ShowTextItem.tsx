import ColoredLine from './ColoredLine';

interface ShowTextItemProps {
  title: string;
}

export default function ShowTextItem({ title }: ShowTextItemProps) {
  return (
    <li className='w-24 min-w-fit cursor-pointer relative text-lg hover:scale-105 mb-3'>
      <button className=''>
        {title}
      </button>
      <ColoredLine
        displayOn='bottom'
      />
    </li>
  );
}
