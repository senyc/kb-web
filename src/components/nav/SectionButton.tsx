interface SectionButtonProps {
  title: string;
  onClick: any;
}

export default function SectionButton({ title, onClick }: SectionButtonProps) {
  return (
    <button 
      className='w-56 hover:scale-105 font-thin text-4xl'
      onClick={onClick}
    >
      {title}
    </button>
  );
}
