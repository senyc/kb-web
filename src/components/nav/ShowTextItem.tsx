import ColoredLine from '@components/main/ColoredLine';

interface ShowTextItemProps {
  title: string;
  onClick: () => void
}

export default function ShowTextItem({ title, onClick }: ShowTextItemProps) {
  return (
    <li className='relative mb-3 w-24 min-w-fit cursor-pointer text-lg hover:scale-105'>
      <button onClick={onClick}>
        {title}
      </button>
      <ColoredLine
        displayOn='bottom'
      />
    </li>
  );
}
