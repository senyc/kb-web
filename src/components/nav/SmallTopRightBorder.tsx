export default function SmallTopRightBorder() {
  return (
    <div className='top-left-border'></div>
  );
}
