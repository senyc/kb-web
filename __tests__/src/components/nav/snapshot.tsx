import SectionButton from '@components/nav/SectionButton';
import ShowTextItem from '@components/nav/ShowTextItem';
import SmallTopRightBorder from '@components/nav/SmallTopRightBorder';
import { create } from 'react-test-renderer';

describe('main elements match recorded snapshots', () => {
  describe('Section Button', () => {
    it('renders the title', () => {
      const tree = create(<SectionButton title='test' onClick={() => undefined} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('SmallToprightBorder', () => {
    it('renders', () => {
      const tree = create(<SmallTopRightBorder />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('ShowTextItem', () => {
    it('renders', () => {
      const tree = create(<ShowTextItem title={'tests'} onClick={() => undefined} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
