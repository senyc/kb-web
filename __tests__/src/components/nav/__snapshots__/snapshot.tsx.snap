// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main elements match recorded snapshots Section Button renders the title 1`] = `
<button
  className=" text-3xl font-thin hover:scale-105 sm:text-4xl"
  onClick={[Function]}
>
  test
</button>
`;

exports[`main elements match recorded snapshots ShowTextItem renders 1`] = `
<li
  className="relative mb-3 w-24 min-w-fit cursor-pointer text-lg hover:scale-105"
>
  <button
    onClick={[Function]}
  >
    tests
  </button>
  <span
    className="styled-vertical-line absolute -bottom-1 left-0 h-px px-11"
  />
</li>
`;

exports[`main elements match recorded snapshots SmallToprightBorder renders 1`] = `
<div
  className="top-left-border"
/>
`;
